### PR TITLE
SR-10469 Unknown syntax for interpolated string literal

### PR DESCRIPTION
**Pull request by kitasuke**
## Overview
This PR fixes unknown syntax for interpolated string literal by changing `Expr` kind of `Expression` in `ExpressionSegment` to `FunctionCallArgumentList`

## Ticket
Resolves #52869.

### CODE REVIEW

File: test/IDE/complete_expr_postfix_begin.swift
Comment (kitasuke) [inline test/IDE/complete_expr_postfix_begin.swift]: I wasn't able to fix this failing test, so changed to `RUN-FIXME` for now. Can someone give me a better solution?

**Comment by rintaro** [comment]
Could you add test cases with multiple arguments with labels? Like `"test \(foo, default: 2)"`.

File: test/IDE/complete_expr_postfix_begin.swift
Comment (rintaro) [reply test/IDE/complete_expr_postfix_begin.swift]: It's OK for now. I'll fix this after merging this.

File: utils/gyb_syntax_support/ExprNodes.py
Comment (rintaro) [inline utils/gyb_syntax_support/ExprNodes.py]: You probably need `collection_element_name` after #24227 .

**Comment by kitasuke** [comment]
Thank you for your feedback! Will add the test cases

**Comment by kitasuke** [comment]
@rintaro Addressed your feedbacks

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test Linux platform

**Comment by rintaro** [comment]
@swift-ci Please smoke test Linux platform

**Review by rintaro — APPROVED** [review]

**Comment by rintaro** [comment]
Thank you!

**Comment by shahmishal** [comment]
Test failure, going to revert to unblock PR testing.

https://ci.swift.org/job/oss-swift-pr-test/7259/changes#detail

```
01:41:07.590 ******************** TEST 'SwiftSyntax :: print_verify_tree.swift' FAILED ********************
01:41:07.590 Script:
01:41:07.590 --
01:41:07.590 : 'RUN: at line 1';   rm -rf "/private/var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/Output/print_verify_tree.swift.tmp" && mkdir -p "/private/var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/Output/print_verify_tree.swift.tmp"
01:41:07.590 : 'RUN: at line 2';   /Users/buildnode/jenkins/workspace/oss-swift-pr-test/buildbot_incremental/swiftsyntax-macosx-x86_64/x86_64-apple-macosx/release/lit-test-helper -print-tree -source-file /Users/buildnode/jenkins/workspace/oss-swift-pr-test/swift-syntax/lit_tests/print_verify_tree.swift > /private/var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/Output/print_verify_tree.swift.tmp.result
01:41:07.590 : 'RUN: at line 3';   diff -u /private/var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/Output/print_verify_tree.swift.tmp.result /Users/buildnode/jenkins/workspace/oss-swift-pr-test/swift-syntax/lit_tests/output/print_verify_tree.swift.withkind
01:41:07.590 --
01:41:07.590 Exit Code: 1
01:41:07.590 
01:41:07.590 Command Output (stdout):
01:41:07.590 --
01:41:07.590 --- /private/var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/Output/print_verify_tree.swift.tmp.result	2019-04-29 20:11:11.000000000 -0700
01:41:07.590 +++ /Users/buildnode/jenkins/workspace/oss-swift-pr-test/swift-syntax/lit_tests/output/print_verify_tree.swift.withkind	2019-04-29 05:37:46.000000000 -0700
01:41:07.590 @@ -4,8 +4,8 @@
01:41:07.590  
01:41:07.590  func </TokenSyntax><TokenSyntax>foo</TokenSyntax><FunctionSignatureSyntax><ParameterClauseSyntax><TokenSyntax>(</TokenSyntax><FunctionParameterListSyntax></FunctionParameterListSyntax><TokenSyntax>) </TokenSyntax></ParameterClauseSyntax></FunctionSignatureSyntax><CodeBlockSyntax><TokenSyntax>{</TokenSyntax><CodeBlockItemListSyntax><CodeBlockItemSyntax><IfConfigDeclSyntax><IfConfigClauseListSyntax><IfConfigClauseSyntax><TokenSyntax>
01:41:07.590  #if </TokenSyntax><FunctionCallExprSyntax><IdentifierExprSyntax><TokenSyntax>swift</TokenSyntax></IdentifierExprSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><PrefixOperatorExprSyntax><TokenSyntax>>=</TokenSyntax><FloatLiteralExprSyntax><TokenSyntax>3.2</TokenSyntax></FloatLiteralExprSyntax></PrefixOperatorExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></FunctionCallExprSyntax><CodeBlockItemListSyntax><CodeBlockItemSyntax><FunctionCallExprSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>
01:41:07.590 -    components</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>append</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><StringInterpolationExprSyntax><TokenSyntax>"</TokenSyntax><StringInterpolationSegmentsSyntax><StringSegmentSyntax><TokenSyntax>-b \"</TokenSyntax></StringSegmentSyntax><ExpressionSegmentSyntax><TokenSyntax>\</TokenSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><SubscriptExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>[</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><PrefixOperatorExprSyntax><TokenSyntax>..<</TokenSyntax><FunctionCallExprSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>index</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><TokenSyntax>before</TokenSyntax><TokenSyntax>: </TokenSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>endIndex</TokenSyntax></MemberAccessExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></FunctionCallExprSyntax></PrefixOperatorExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>]</TokenSyntax></SubscriptExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></ExpressionSegmentSyntax><StringSegmentSyntax><TokenSyntax>\"</TokenSyntax></StringSegmentSyntax></StringInterpolationSegmentsSyntax><TokenSyntax>"</TokenSyntax></StringInterpolationExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></FunctionCallExprSyntax></CodeBlockItemSyntax></CodeBlockItemListSyntax></IfConfigClauseSyntax><IfConfigClauseSyntax><TokenSyntax>
01:41:07.590 +    components</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>append</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><StringInterpolationExprSyntax><TokenSyntax>"</TokenSyntax><StringInterpolationSegmentsSyntax><StringSegmentSyntax><TokenSyntax>-b \"</TokenSyntax></StringSegmentSyntax><ExpressionSegmentSyntax><TokenSyntax>\</TokenSyntax><TokenSyntax>(</TokenSyntax><SubscriptExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>[</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><PrefixOperatorExprSyntax><TokenSyntax>..<</TokenSyntax><FunctionCallExprSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>index</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><TokenSyntax>before</TokenSyntax><TokenSyntax>: </TokenSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>endIndex</TokenSyntax></MemberAccessExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></FunctionCallExprSyntax></PrefixOperatorExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>]</TokenSyntax></SubscriptExprSyntax><TokenSyntax>)</TokenSyntax></ExpressionSegmentSyntax><StringSegmentSyntax><TokenSyntax>\"</TokenSyntax></StringSegmentSyntax></StringInterpolationSegmentsSyntax><TokenSyntax>"</TokenSyntax></StringInterpolationExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></FunctionCallExprSyntax></CodeBlockItemSyntax></CodeBlockItemListSyntax></IfConfigClauseSyntax><IfConfigClauseSyntax><TokenSyntax>
01:41:07.590  #else</TokenSyntax><CodeBlockItemListSyntax><CodeBlockItemSyntax><FunctionCallExprSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>
01:41:07.590 -    components</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>append</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><StringInterpolationExprSyntax><TokenSyntax>"</TokenSyntax><StringInterpolationSegmentsSyntax><StringSegmentSyntax><TokenSyntax>-b \"</TokenSyntax></StringSegmentSyntax><ExpressionSegmentSyntax><TokenSyntax>\</TokenSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><FunctionCallExprSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>substring</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><TokenSyntax>to</TokenSyntax><TokenSyntax>: </TokenSyntax><FunctionCallExprSyntax><MemberAccessExprSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>characters</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>index</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><TokenSyntax>before</TokenSyntax><TokenSyntax>: </TokenSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>endIndex</TokenSyntax></MemberAccessExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></FunctionCallExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></FunctionCallExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></ExpressionSegmentSyntax><StringSegmentSyntax><TokenSyntax>\"</TokenSyntax></StringSegmentSyntax></StringInterpolationSegmentsSyntax><TokenSyntax>"</TokenSyntax></StringInterpolationExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></FunctionCallExprSyntax></CodeBlockItemSyntax></CodeBlockItemListSyntax></IfConfigClauseSyntax></IfConfigClauseListSyntax><TokenSyntax>
01:41:07.590 +    components</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>append</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><StringInterpolationExprSyntax><TokenSyntax>"</TokenSyntax><StringInterpolationSegmentsSyntax><StringSegmentSyntax><TokenSyntax>-b \"</TokenSyntax></StringSegmentSyntax><ExpressionSegmentSyntax><TokenSyntax>\</TokenSyntax><TokenSyntax>(</TokenSyntax><FunctionCallExprSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>substring</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><TokenSyntax>to</TokenSyntax><TokenSyntax>: </TokenSyntax><FunctionCallExprSyntax><MemberAccessExprSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>characters</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>index</TokenSyntax></MemberAccessExprSyntax><TokenSyntax>(</TokenSyntax><FunctionCallArgumentListSyntax><FunctionCallArgumentSyntax><TokenSyntax>before</TokenSyntax><TokenSyntax>: </TokenSyntax><MemberAccessExprSyntax><IdentifierExprSyntax><TokenSyntax>string</TokenSyntax></IdentifierExprSyntax><TokenSyntax>.</TokenSyntax><TokenSyntax>endIndex</TokenSyntax></MemberAccessExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></FunctionCallExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></FunctionCallExprSyntax><TokenSyntax>)</TokenSyntax></ExpressionSegmentSyntax><StringSegmentSyntax><TokenSyntax>\"</TokenSyntax></StringSegmentSyntax></StringInterpolationSegmentsSyntax><TokenSyntax>"</TokenSyntax></StringInterpolationExprSyntax></FunctionCallArgumentSyntax></FunctionCallArgumentListSyntax><TokenSyntax>)</TokenSyntax></FunctionCallExprSyntax></CodeBlockItemSyntax></CodeBlockItemListSyntax></IfConfigClauseSyntax></IfConfigClauseListSyntax><TokenSyntax>
01:41:07.590  #endif</TokenSyntax></IfConfigDeclSyntax></CodeBlockItemSyntax></CodeBlockItemListSyntax><TokenSyntax>
01:41:07.590  }</TokenSyntax></CodeBlockSyntax></FunctionDeclSyntax></CodeBlockItemSyntax></CodeBlockItemListSyntax><TokenSyntax></TokenSyntax></SourceFileSyntax>
01:41:07.590 \ No newline at end of file
01:41:07.590 
01:41:07.590 --
01:41:07.590 
01:41:07.590 ********************

```

**Comment by kitasuke** [comment]
Sorry about the failed test. I'm working on the fix and will create a new PR.

**Pull request merged**
